### PR TITLE
Fix AttributeError: 'LayerList' object has no attribute 'name'

### LINKED ISSUE #4271: AttributeError: 'LayerList' object has no attribute 'name'
## 🐛 Bug

- open napari, 
- Click on empty canvas.
- Press Ctrl-C
Console will show:

```
File ~/dev/napari/napari/_qt/containers/qt_layer_model.py:17, in QtLayerListModel.data(self=<napari._qt.containers.qt_layer_model.QtLayerListModel object>, index=<PyQt5.QtCore.QModelIndex object>, role=0)
     16 if role == Qt.DisplayRole:  # used for item text
---> 17     return layer.name
        layer = []
     18 if role == Qt.TextAlignmentRole:  # alignment of the text
     19     return Qt.AlignCenter

AttributeError: 'LayerList' object has no attribute 'name'
```

## Expected behavior

Don't.

## Environment

This seem to be because `_base_item_model:_BaseEventedItemModel.getItem` is not type stable.

```
    def getItem(self, index: QModelIndex) -> ItemType:
        """Return python object for a given `QModelIndex`.

        An invalid `QModelIndex` will return the root object.
        """
        return self._root[index.row()] if index.isValid() else self._root
```

The `else_self.root` is not of Type `ItemType`, but an (Empty) container.

### PR DESCRIPTION
**Pull request by tlambert03**
# Description 
fixes #4271 
(actually, that may have already been fixed by #4480 ... but this can't hurt either)

@Carreau, I actually wasn't able to reliably reproduce this.  I _did_ get it to happen once after a number of launches and key-mashing, so it's definitely there... but i can't figure out what's going on with keypresses that makes it indeterminate for me.

Anyway, I think this should fix it.  If you have a moment could you test?

### CODE REVIEW

File: napari/_qt/containers/_layer_delegate.py [hunk 1]
```diff
@@ -92,6 +92,8 @@ def paint(
     def get_layer_icon(self, option, index):
         """Add the appropriate QIcon to the item based on the layer type."""
         layer = index.data(ItemRole)
+        if layer is None:
+            return
```
Comment (Carreau) [inline napari/_qt/containers/_layer_delegate.py:95]: Added this.

My main concern is if we respect the API of QAbstractModelItem:

https://doc.qt.io/qtforpython-5/PySide2/QtCore/QAbstractItemModel.html#PySide2.QtCore.PySide2.QtCore.QAbstractItemModel.data

Comment (tlambert03) [reply napari/_qt/containers/_layer_delegate.py:95]: are you referring to the invalid QVariant return type?  If so, `None` is the invalid QVariant for python bindings:
https://doc.qt.io/archives/qtforpython-5.12/pysideapi2.html#qvariant

Comment (Carreau) [reply napari/_qt/containers/_layer_delegate.py:95]: Ah, yes, perfect then.

Comment (tlambert03) [reply napari/_qt/containers/_layer_delegate.py:95]: thanks for adding this bit!

**Comment by Carreau** [comment]
Seem a couple of places need a 'not None' check. Otherwise we can also just check that the "layer" object hasattr('name').

**Comment by tlambert03** [comment]
arg... lots of fails.  will have a look later today

**Comment by andy-sweet** [comment]
@tlambert03 : I was looking through some bugs to fix and found #4271 . Are you or @Carreau actively working on this PR? Or planning to soon? If not, is it worth me taking and trying to get it across the line?

**Comment by tlambert03** [comment]
Would be happy to have help!  Shouldn't be much I hope

**Comment by jni** [comment]
Is this a regression from 0.4.15 or is it older? ie should this go in the 0.4.16 milestone?

**Comment by andy-sweet** [comment]
I'm not sure. I haven't actually reproduced the bug yet described in #4271. I plan to take a look tomorrow and will try to reproduce it, ideally with a test, and will add that info to the issue.

**Comment by tlambert03** [comment]
It's older, and actually it might also relate to #4421 and #4425 (that is, the best fix might be dealing with the order/handling of event emission)

**Comment by tlambert03** [comment]
this is ready to go, and has turned into a very minor "just in case" bugfix.  with the real fix probably having come from #4480

**Review by jni — APPROVED** [review]

**Review by goanpeca — APPROVED** [review]

**Review by Czaki — APPROVED** [review]

**Pull request merged**